### ISSUE TITLE
Handle internal error in `api/v2/status.py` endpoint

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1446]
```
Traceback (most recent call last):
  File "/somewhere/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/somewhere/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/somewhere/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/somewhere/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/somewhere/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/somewhere/lib/python3.10/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/somewhere/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/somewhere/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/somewhere/lib/python3.10/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
  File "/somewhere/lib/python3.10/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
  File "/somewhere/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/somewhere/lib/python3.10/site-packages/fractal_server/app/routes/api/v2/status.py", line 116, in get_workflowtask_status
    last_valid_wftask_id = workflow.task_list[end - 1]
IndexError: list index out of range
```

**Comment by tcompa** [comment]
Side remark: the line `last_valid_wftask_id = workflow.task_list[end - 1]` is also wrong for another reason (missing `.id`).
Ref:
* 7fafa5701c487e71010306ec7aaf87d9704d08ea
* b04b7970d20c87978019e945652e677db6738da5

**Comment by tcompa** [comment]
The only way I found to reproduce the internal-error in the first comment is like the following:
1. Create a workflow with 3 workflow tasks
2. Create a submitted job which runs tasks 0 and 1
3. Remove tasks 1 and 2 from the workflow
4. Call the status endpoint

(ref github.com/fractal-analytics-platform/fractal-server/pull/1449/commits/e0f92dbf1e3c117b3d9f375b909be7235e9ca86f)

This means that we are out-of-scope with respect to the status endpoint (see https://github.com/fractal-analytics-platform/fractal-server/issues/1083#issuecomment-2063898906), and we cannot prepare a fully consistent response.
The fix in #1449 handles the internal error (and it shows some relevant warnings in the server logs), and then it simply sets `last_valid_wftask_id` to `None`.

**Comment by jluethi** [comment]
> Remove tasks 1 and 2 from the workflow

Ah, I think I did this during some testing! Not really intending to test the behavior, just already modifying a workflow while it was running, as the last task wasn't actually needed.

> The fix in https://github.com/fractal-analytics-platform/fractal-server/pull/1449 handles the internal error (and it shows some relevant warnings in the server logs), and then it simply sets last_valid_wftask_id to None.

Sounds good to me
